### PR TITLE
update to version 12.1. Use mingw version of pthreads

### PR DESCRIPTION
**Pull request by andreas-aur**

### CODE REVIEW

**Comment by andreas-aur** [comment]
update postgres to version 12.1
this required changes to to the 0001 dll patch
the 0002 patch  was added to use the mingw cross compiler pthreads instead of the postgres wrappers

File: postgresql/mingw-w64/PKGBUILD [hunk 1]
```diff
@@ -43,7 +46,7 @@ build() {
     ln -s "${srcdir}/postgresql-$pkgver/src/interfaces/ecpg/pgtypeslib/libpgtypesdll.def" src/interfaces/ecpg/pgtypeslib/
     ln -s "${srcdir}/postgresql-$pkgver/src/interfaces/ecpg/compatlib/libecpg_compatdll.def" src/interfaces/ecpg/compatlib/
 
-    make
+    make -j4
```
Comment (Martchus) [inline postgresql/mingw-w64/PKGBUILD]: Please don't. Modify your `makepkg.conf` to customize those options.

Comment (andreas-aur) [reply postgresql/mingw-w64/PKGBUILD]: OK

**Comment by andreas-aur** [comment]
Do we need the dll patch? 
I've compiled and linked it statically without the dll patch - this works fine for me. 
One quirk was that I needed to link with -lpq -lpgcommon -lpgport

**Comment by Martchus** [comment]
The build passed. I'm merging the PR soon and will push the commit to the AUR.

The import library still has `*.dll.a` suffix so you rebased the patch correctly. And yes, for consistency with other libraries it is nice to use the `*.dll.a` suffix here as well. Of course this is not relevant for you when you use the static version. And yes, in this case you need to specify all transitive dependencies in the right order. I did the same for the static Qt builds. Unfortunately the pkg-config file doesn't help one with that as it is apparently only supposed to be used for dynamic linking. (That caveat didn't change compared to postgresl 11.)

**Comment by Martchus** [comment]
Merged locally and also pushed to the AUR.

**Pull request closed without merging**